src/core/crypto/nacl_catapult.ts

```diff
@@ -850,3 +850,57 @@ export const unpackneg = (r, p) => {
     M(r[3], r[0], r[1]);
     return 0;
 };
+
+export const unpack = (r, p) => {
```

Comment:
please explain the need to replace `unpackneg` with this ?

Reply:
TBH, I haven't figure out why yet. Have been asked core devs on slack. 

Basically, this is only the difference from NIS1 code. We get different hash (last byte) by using 'unpackneg()`, but using `unpack()` works fine. Not quite sure why. The code looks the same to me

Reply:
for shared key derivation, `unpack` must be used to be compliant with catapult test vectors.